kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/cloudconnection/factory/CloudConnectionManagerFactory.java

```diff
@@ -0,0 +1,112 @@
+
+    /**
+     * Returns the factory PID of the OSGi Factory Component represented by this {@link CloudConnectionManager}.
+     *
+     * @return a String representing the factory PID of the Factory Component.
+     */
+    public String getFactoryPid();
+
+    /**
+     * Creates a {@link CloudConnectionManager} instance and initializes its configuration with the defaults
+     * expressed in the Metatype of the target component factory providing the {@link CloudConnectionManager}.
+     *
+     * The created {@link CloudConnectionManager} instance will have its {@code kura.service.pid} property
+     * set to the value provided in the {@code pid} parameter.
+     *
+     * @param pid
+     *            the Kura persistent identifier ({@code kura.service.pid}) of the {@link CloudConnectionManager}
+     *            instance created by this factory.
+     * @throws KuraException
+     *             an exception is thrown in case the creation operation fails
+     */
+    public void createConfiguration(String pid) throws KuraException;
+
+    /**
+     * Returns the list of {@code kura.service.pid}s that compose the cloud stack associated with the provided
+     * {@code kura.service.pid} associated to a {@link CloudConnectionManager} instance.
+     *
+     * @param pid
+     *            the Kura persistent identifier, {@code kura.service.pid}, of a {@link CloudConnectionManager}
+     *            instance.
+     * @return the {@link List} of {@code kura.service.pid}s related to the provided {@code pid}.
+     * @throws KuraException
+     *             if the specified {@code kura.service.pid} is incorrect.
+     */
+    public List<String> getStackComponentsPids(String pid) throws KuraException;
+
+    /**
+     * Deletes a previously created configuration deactivating the associated {@link CloudConnectionManager} instance.
+     *
+     * @param pid
+     *            the Kura persistent identifier, {@code kura.service.pid}, of the {@link CloudConnectionManager}
+     *            instance.
+     * @throws KuraException
+     *             if the provided {@code kura.service.pid} is incorrect or the delete operation fails.
+     */
+    public void deleteConfiguration(String pid) throws KuraException;
+
+    /**
+     * Returns a set of services managed by this factory
+     * It is up to the factory to specify how to assembles the result. The PIDs returned by this list must be the PIDs
+     * assigned
+     * to the OSGi service property {@code kura.service.pid} and it must be possible to pass all those results into
+     * the method {@link #getStackComponentsPids(String)} of the same factory instance.
+     *
+     * The IDs returned by this method must not necessarily point to registered OSGi services. But if they do, they must
+     * point only to instances of {@link CloudConnectionManager}.
+     *
+     * @return the set of services, never returns {@code null}
```

Comment:
instead of never returns {@code null}, it would be more conforming to API if we write -> 

@return the set of services or an empty set